packages/components/src/data-model/word-query/lexical-data/data-objects/treebank-data.js

```diff
@@ -0,0 +1,64 @@
+import LexicalDataResult from '@comp/data-model/word-query/lexical-data/result/lexical-data-result.js'
+import { LanguageModelFactory as LMF, Logger, HomonymGroup } from 'alpheios-data-models'
+import { ClientAdapters } from 'alpheios-client-adapters'
+import WordQueryErrorCodes from '@comp/data-model/word-query/error/word-query-error-codes.js'
+import ErrorMapper from '@comp/data-model/word-query/error/error-mapper.js'
+
+export default class TreebankData {
+  constructor ({ word, language, clientId, treebankProvider, treebankSentenceId, treebankWordIds } = {}) {
+    this._word = word
+    this._language = language
+    this._clientId = clientId
+    this._treebankProvider = treebankProvider
+    this._treebankSentenceId = treebankSentenceId
+    this._treebankWordIds = treebankWordIds
+  }
+
+  async retrieve () {
```

Comment:
since the retrieve call is always passed a lexical data object, maybe it should be here even though it's not used (and will be empty)?

Reply:
I'm not sure what are best practices in cases like this, but I think it makes sense. I will add a comment explaining why this variable is not used.

I also think it would make sense to create a common base class for all `Data` objects in order to define their common interface. What's your opinion on that?